### PR TITLE
[DXP-647] Publishing data

### PR DESCRIPTION
**Pull request by jgardo-ds**

### CODE REVIEW

File: src/main/java/dev/streamx/cli/publish/payload/PayloadException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+
+public class PayloadException extends RuntimeException {
+  private PayloadException(String prefix, Exception exception) {
+    super(prefix + exception.getMessage(), exception);
+  }
+
+  public static PayloadException jsonProcessingException(JsonProcessingException exception) {
+    return new PayloadException("Json processing exception. ", exception);
```
Comment (michalcukierman) [inline src/main/java/dev/streamx/cli/publish/payload/PayloadException.java:12]: JSON

File: src/main/java/dev/streamx/cli/publish/PublishCommand.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+@Command(name = "publish", mixinStandardHelpOptions = true)
+public class PublishCommand implements Runnable {
+
+  @Parameters(index = "0", description = "Channel that message will be published to")
+  String channel;
+
+  @Parameters(index = "1", description = "Message key")
+  String key;
+
+  @Option(names = "--ingestion-url",
```
Comment (michalcukierman) [inline src/main/java/dev/streamx/cli/publish/PublishCommand.java:28]: as an improvement, we could read it from .env file or properties

File: src/main/java/dev/streamx/cli/util/ExceptionUtils.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package dev.streamx.cli.util;
+
+public final class ExceptionUtils {
+  public static RuntimeException sneakyThrow(Throwable t) {
+    if (t == null) {
+      throw new NullPointerException("t");
```
Comment (michalcukierman) [inline src/main/java/dev/streamx/cli/util/ExceptionUtils.java:6]: should we pass a string in a constructor?

File: src/main/java/dev/streamx/cli/publish/PublishCommand.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  PayloadResolver payloadResolver;
+
+  @Override
+  public void run() {
+    provideSchema();
+
+    JsonNode jsonNode = payloadResolver.createPayload(data);
+
+    try (StreamxClient client = streamxClientProvider.createStreamxClient()) {
+      var pagePublisher = client.newPublisher(channel, JsonNode.class);
```
Comment (kamilchociej) [inline src/main/java/dev/streamx/cli/publish/PublishCommand.java:63]: why pagePublisher?

File: src/main/java/dev/streamx/cli/publish/PublishCommand.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  @Inject
+  IngestionClientContext ingestionClientContext;
+
+  @Inject
+  PayloadResolver payloadResolver;
+
+  @Override
+  public void run() {
+    provideSchema();
```
Comment (kamilchociej) [inline src/main/java/dev/streamx/cli/publish/PublishCommand.java:58]: Provided schema is ignored. Please change this method name or use returned object.

File: src/main/java/dev/streamx/cli/publish/payload/PayloadException.java (hunk 1 above)
Comment (kamilchociej) [inline src/main/java/dev/streamx/cli/publish/payload/PayloadException.java:12]: If this message for user then it should give more details and hint how to proceed in such situation.

File: src/test/java/dev/streamx/cli/publish/PublishCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  public void shouldHandleBadRequestFromRestIngestionApi(QuarkusMainLauncher launcher) {
+    // when
+    LaunchResult result = launcher.launch("publish",
+        "--ingestion-url=" + getIngestionUrl(),
+        "--data=" + DATA,
+        BAD_REQUEST_CHANNEL, KEY);
+
+    // then
+    assertThat(result.exitCode()).isNotZero();
+    assertThat(result.getErrorOutput()).contains("Ingestion client exception. Message: Publication Ingestion REST endpoint known error. Code: INVALID_PUBLICATION_PAYLOAD. Message: Error message");
```
Comment (kamilchociej) [inline src/test/java/dev/streamx/cli/publish/PublishCommandTest.java:85]: Why do we have "Message:" twice in this output?

File: src/main/java/dev/streamx/cli/publish/PublishCommand.java (hunk 2 above)
Comment (jgardo-ds) [reply src/main/java/dev/streamx/cli/publish/PublishCommand.java:28]: There are multiple posibilities. I was thinking about `namespace` concepts like in kubectl. And I think they may be conflicting approaches.

Moreover in current scope we usualy follow conventions, that ingestion url is localhost:8080 (in this task, we focus on publishing to local mesh).
Enabling other environments should be done in following tasks.

File: src/main/java/dev/streamx/cli/util/ExceptionUtils.java (hunk 3 above)
Comment (jgardo-ds) [reply src/main/java/dev/streamx/cli/util/ExceptionUtils.java:6]: It's copy-pasted from [Lombok](https://github.com/projectlombok/lombok/blob/master/src/core/lombok/Lombok.java#L28)

As we do not use Lombok in our project (maybe we should?) I 'imported' it's utils mechanism allowing to throw checked exceptions as uncheck (as Sneaky throws it's well known mechanism and widely used)

File: src/test/java/dev/streamx/cli/publish/PublishCommandTest.java (hunk 6 above)
Comment (jgardo-ds) [reply src/test/java/dev/streamx/cli/publish/PublishCommandTest.java:85]: Yes, double "Message" will be fixed.

However Messages from StreamxClientExceptions requires more work that will be done within [DXP-707](https://teamds.atlassian.net/browse/DXP-707)

**Pull request closed without merging**